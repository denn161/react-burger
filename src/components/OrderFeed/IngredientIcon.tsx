import React from 'react'
import { ingredientsSelector } from '../../services/selectors/ingredientsSelector'
import { useSelector } from '../../services/store/hooks'
import styles from './OrderFild.module.scss'

interface IIngridientIconProps {
  img?: string

}

const IngredientIcon = ({ img }: IIngridientIconProps) => {

  return (
    <>
      <img className={styles.image} data-testid='image-tag' src={img} alt="Ingredient" />
    </>


  )
}

export default IngredientIcon