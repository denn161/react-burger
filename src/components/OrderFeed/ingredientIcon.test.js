import { cleanup, getByAltText, getByTestId, render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import IngredientIcon from './IngredientIcon'

describe('test IngredientIcon component', () => {

    const mockPropUrl = 'https://avatars.mds.yandex.net/i?id=1834c191b774e63cde2835ef7cc00976a6e6489b-8425275-images-thumbs&n=13'


    test('should be equal to snapshot', () => {

        const renderValue = renderer
            .create(
                <IngredientIcon img={mockPropUrl} />
            ).toJSON()

        expect(renderValue).toMatchSnapshot()
    })


})

describe('Testing component IngredientIcon witch render', () => {

    afterEach(cleanup)

    const mockPropUrl = 'https://avatars.mds.yandex.net/i?id=1834c191b774e63cde2835ef7cc00976a6e6489b-8425275-images-thumbs&n=13'
    const mockAlt = 'Ingredient'

    test('should render IngredientIcon with class image', () => {

        const { container } = render(

            <IngredientIcon img={mockPropUrl} />
        )

        const img = getByAltText(container, mockAlt)

        expect(img).toHaveClass('image')



    })

    test('should render IngredientIcon with class image', () => {

        const { container } = render(

            <IngredientIcon img={mockPropUrl} />
        )

        const img = getByTestId(container,'image-tag')

        expect(img).toHaveClass('image')



    })

})