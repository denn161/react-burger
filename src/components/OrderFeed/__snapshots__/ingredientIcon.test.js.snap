// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test IngredientIcon component should be equal to snapshot 1`] = `
<img
  alt="Ingredient"
  className="image"
  data-testid="image-tag"
  src="https://avatars.mds.yandex.net/i?id=1834c191b774e63cde2835ef7cc00976a6e6489b-8425275-images-thumbs&n=13"
/>
`;
